src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +238,155 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void CheckForReplacedByLinks(string parentPatientId, SearchResultEntry parentPatientResource)
+        {
+            List<Patient.LinkComponent> links = ExtractLinksFromParentPatient(parentPatientResource);
+
+            if (links == null)
+            {
+                return;
+            }
+
+            // If a "replaced-by" link is present, it indicates that the patient resource containing this link must no
+            // longer be used. The link points forward to another patient resource that must be used in lieu of the
+            // resource that contains the "replaced-by" link.
+            foreach (Patient.LinkComponent link in links)
+            {
+                ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                // Regardless of the other links present, we throw an error if we find a "replaced-by" link.
+                if (link.Type == Patient.LinkType.ReplacedBy)
+                {
+                    // Specify the url that must be used instead.
+                    Uri url = _urlResolver.ResolveOperationResultUrl(OperationsConstants.PatientEverything, referenceSearchValue.ResourceId);
+
+                    throw new EverythingOperationException(
+                        string.Format(
+                            Core.Resources.EverythingOperationResourceIrrelevant,
+                            parentPatientId,
+                            referenceSearchValue.ResourceId),
+                        HttpStatusCode.MovedPermanently,
+                        url.ToString());
```

Comment:
Good question. I assumed that we could simply return an error as soon as we encounter a `replaced-by` link, since it doesn't make much sense for a patient to have multiple. However, from reading through the spec, it doesn't seem like there is anything that would stop someone from doing so.

One thought: we could return a separate error message if a patient has more than one `replaced-by` link. @CaitlinV39 thoughts?

Reply:
I'm fine with whatever is simplest. I think it would be hard to know which would be the "right one". The **best** option is probably to say there are multiple, but I think passing the first one and then if they get another error thats okay. It sounds like a business process inefficiency if a single patient has multiple replaced-by links.

Reply:
Sounds good, will keep as-is for now. I created a backlog item that we can prioritize later: [AB#85435](https://microsofthealth.visualstudio.com/Health/_workitems/edit/85435).